admin/urls.go

```diff
@@ -106,6 +106,15 @@ func (u *URLs) IsHTTPS() bool {
 	return u.https
 }
 
+// IsCustomDomain returns true if the given domain is a custom domain.
+func (u *URLs) IsCustomDomain(domain string) bool {
+	externalURL, err := url.Parse(u.external)
+	if err != nil {
+		panic(fmt.Errorf("failed to parse external domain %q: %w", u.external, err))
+	}
+	return !strings.EqualFold(externalURL.Host, domain)
+}
```

Comment:
Do you think it would make sense to also parse the domain just to be sure there's no scheme or trailing `/`? 

```suggestion
// IsCustomDomain returns true if the given domain is a custom domain.
func (u *URLs) IsCustomDomain(domain string) bool {
	externalURL, err := url.Parse(u.external)
	if err != nil {
		panic(fmt.Errorf("failed to parse external domain %q: %w", u.external, err))
	}
	domainURL, err := url.Parse(domain)
	if err != nil {
		panic(fmt.Errorf("failed to parse provided domain %q: %w", domain, err))
	}
	return !strings.EqualFold(externalURL.Host, domainURL.Host)
}
```